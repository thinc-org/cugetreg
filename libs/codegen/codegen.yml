overwrite: true
schema: apps/cugetreg-api/**/*.graphql
documents: libs/codegen/src/query/*.graphql

generates:
  libs/codegen/src/generated/index.tsx:
    plugins:
      - typescript
      - typescript-operations
      - typescript-react-apollo
    config:
      withHooks: true
      useTypeImports: true
      gqlImport: graphql-tag
