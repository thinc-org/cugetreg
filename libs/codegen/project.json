{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/codegen/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": [
        "{options.outputFile}"
      ],
      "options": {
        "lintFilePatterns": [
          "libs/codegen/**/*.ts"
        ]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": [
        "coverage/libs/codegen"
      ],
      "options": {
        "jestConfig": "libs/codegen/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "codegen": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          "graphql-codegen --config libs/codegen/codegen.yml"
        ]
      }
    }
  },
  "tags": []
}