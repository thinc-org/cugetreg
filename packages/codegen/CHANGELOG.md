# @cgr/codegen

## 1.2.1-beta.0

### Patch Changes

- 8d9d304b: bump eslint and fix ci

## 1.2.0

### Minor Changes

- 7be734a2: make exam period and period fields nullable in graphql

### Patch Changes

- 7be734a2: use graphql schemas from apps/api/src directory
- 7be734a2: remove calendarId stuffs and fix type in course cart

## 1.2.0-beta.0

### Minor Changes

- 7be734a2: make exam period and period fields nullable in graphql

### Patch Changes

- 7be734a2: use graphql schemas from apps/api/src directory
- 7be734a2: remove calendarId stuffs and fix type in course cart

## 1.1.0

### Minor Changes

- fe018c1e: make codegen compatible with turbo

## 1.1.0-beta.0

### Minor Changes

- fe018c1e: make codegen compatible with turbo

## 1.0.0

### Major Changes

- aecb4781: 4 Feb 2023, we're excited to announce the release of the new version of the course registration system, CU Get Reg. This release includes a number of new features, including:

  - Searching for courses by course code or course name
  - Planning your schedule with a course planner
  - Migrate to Turborepo for monorepo management
  - Scrape course data from Reg Chula and Office of Academic Affairs
  - Course recommendation based on your selected course history
  - Review and rate the courses
  - Codegen for generating GraphQL schema and TypeScript types
  - Shared ESLint and Typescript config for consistent code style
