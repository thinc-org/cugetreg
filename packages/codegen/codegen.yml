overwrite: true
schema: ../../apps/api/src/**/*.graphql
documents: ../../packages/codegen/src/query/*.graphql

generates:
  ../../packages/codegen/src/generated/index.tsx:
    plugins:
      - typescript
      - typescript-operations
      - typescript-react-apollo
    config:
      withHooks: true
      useTypeImports: true
      gqlImport: graphql-tag
