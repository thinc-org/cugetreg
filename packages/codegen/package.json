{
  "name": "@cgr/codegen",
  "version": "1.0.0",
  "private": true,
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "scripts": {
    "lint": "eslint '{src,apps,libs,test}/**/*.ts*' --fix"
  },
  "dependencies": {
    "@apollo/client": "3.6.9",
    "graphql-tag": "2.12.6"
  },
  "devDependencies": {
    "eslint-config-custom": "workspace:*",
    "eslint": "7.32.0",
    "tsconfig": "workspace:*"
  }
}
