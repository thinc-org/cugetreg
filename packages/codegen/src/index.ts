export * from './generated'

export * from './future'
