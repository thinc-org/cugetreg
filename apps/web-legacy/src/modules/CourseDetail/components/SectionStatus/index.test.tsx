import { shallow } from 'enzyme'
import { describe, expect, it } from 'vitest'

import { Capacity } from '@cgr/codegen'

describe('SectionStatus', () => {
  const mockCapacity: Capacity = {
    current: 100,
    max: 150,
  }
  const mockFullCapacity: Capacity = {
    current: 151,
    max: 150,
  }

  it.each`
    closed   | capacity
    ${false} | ${mockCapacity}
    ${false} | ${mockFullCapacity}
    ${true}  | ${mockCapacity}
    ${true}  | ${mockFullCapacity}
  `("Should match snapshot correctly'", async ({ closed, capacity }) => {
    const { SectionStatus } = await import('.')

    const wrapper = shallow(<SectionStatus capacity={capacity} closed={closed} />)
    expect(wrapper).toMatchSnapshot()
  })
})
