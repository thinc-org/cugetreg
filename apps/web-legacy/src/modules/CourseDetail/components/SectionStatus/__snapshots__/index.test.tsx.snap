// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SectionStatus > Should match snapshot correctly' 1`] = `
<ForwardRef(Typography)
  ml={1}
  px={2}
  py={0.5}
  sx={
    {
      "backgroundColor": "highlight.green.300",
      "borderRadius": 1,
      "color": "highlight.green.700",
      "height": "fit-content",
      "whiteSpace": "nowrap",
    }
  }
  variant="h6"
>
  <ForwardRef(Grid)
    alignItems="center"
    direction="row"
    spacing={0.5}
  >
    <div>
      available
    </div>
    <NoSeatIcon
      sx={
        {
          "color": "primary.main",
        }
      }
    />
  </ForwardRef(Grid)>
</ForwardRef(Typography)>
`;

exports[`SectionStatus > Should match snapshot correctly' 2`] = `
<ForwardRef(Typography)
  ml={1}
  px={2}
  py={0.5}
  sx={
    {
      "backgroundColor": "highlight.red.300",
      "borderRadius": 1,
      "color": "highlight.red.700",
      "height": "fit-content",
      "whiteSpace": "nowrap",
    }
  }
  variant="h6"
>
  <ForwardRef(Grid)
    alignItems="center"
    direction="row"
    spacing={0.5}
  >
    <div>
      full
    </div>
    <NoSeatIcon
      sx={
        {
          "color": "primary.main",
        }
      }
    />
  </ForwardRef(Grid)>
</ForwardRef(Typography)>
`;

exports[`SectionStatus > Should match snapshot correctly' 3`] = `
<ForwardRef(Typography)
  ml={1}
  px={2}
  py={0.5}
  sx={
    {
      "backgroundColor": "highlight.red.500",
      "borderRadius": 1,
      "color": "white",
      "height": "fit-content",
      "whiteSpace": "nowrap",
    }
  }
  variant="h6"
>
  <ForwardRef(Grid)
    alignItems="center"
    direction="row"
    spacing={0.5}
  >
    <div>
      closed
    </div>
    <NoSeatIcon
      sx={
        {
          "color": "white",
        }
      }
    />
  </ForwardRef(Grid)>
</ForwardRef(Typography)>
`;

exports[`SectionStatus > Should match snapshot correctly' 4`] = `
<ForwardRef(Typography)
  ml={1}
  px={2}
  py={0.5}
  sx={
    {
      "backgroundColor": "highlight.red.500",
      "borderRadius": 1,
      "color": "white",
      "height": "fit-content",
      "whiteSpace": "nowrap",
    }
  }
  variant="h6"
>
  <ForwardRef(Grid)
    alignItems="center"
    direction="row"
    spacing={0.5}
  >
    <div>
      closed
    </div>
    <NoSeatIcon
      sx={
        {
          "color": "white",
        }
      }
    />
  </ForwardRef(Grid)>
</ForwardRef(Typography)>
`;
